src/MediaWiki.ts

```diff
@@ -10,27 +10,37 @@ import semver from 'semver'
 import basicURLDirector from './util/builders/url/basic.director.js'
 import BaseURLDirector from './util/builders/url/base.director.js'
 import ApiURLDirector from './util/builders/url/api.director.js'
+import DesktopURLDirector from './util/builders/url/desktop.director.js'
+import VisualEditorURLDirector from './util/builders/url/visual-editor.director.js'
+import { checkApiAvailabilty } from './util/mw-api.js'
 
 class MediaWiki {
   public metaData: MWMetaData
   public readonly baseUrl: URL
-  public readonly modulePath: string
-  public readonly webUrl: URL
-  public readonly apiUrl: URL
-  public readonly veApiUrl: URL
-  public readonly restApiUrl: URL
-  public readonly mobileRestApiUrl: URL
-  public readonly desktopRestApiUrl: URL
   public readonly getCategories: boolean
+  public readonly modulePathConfig: string
   public readonly namespaces: MWNamespaces = {}
   public readonly namespacesToMirror: string[] = []
 
   private readonly wikiPath: string
+  private readonly restApiPath: string
   private readonly username: string
   private readonly password: string
   private readonly apiPath: string
   private readonly domain: string
   private apiUrlDirector: ApiURLDirector
+  private wikimediaDesktopUrlDirector: DesktopURLDirector
+  private visualEditorURLDirector: VisualEditorURLDirector
+
+  public visualEditorApiUrl: URL
+  public restApiUrl: URL
+  public apiUrl: URL
+  public modulePath: string
+  public webUrl: URL
+  public desktopRestApiUrl: URL
+
+  public hasWikimediaDesktopRestApi = false
+  public hasVisualEditorApi = false
```

Comment:
Looks like the work done on #1878 has not been integrated, so I'm fully lost about how both PRs interact together, nothing seems written about that... but to me they definilty works together.

Reply:
I've integrated the replacement of mw capabilities already.